### PR TITLE
Fix delete branches

### PR DESCRIPTION
**Pull request by clee2000**
Due to PR_WINDOW, if the magic string exists in the body but the pr was not updated recently, the query wouldn't find it and would delete the branch.  Instead, query separately for branches with the no-delete-branch label, which I created recently.

Might as well query for branches with open PRs while we're at it so PRs with the stale label won't get their branches deleted either

### CODE REVIEW

File: .github/scripts/delete_old_branches.py [hunk 1]
```diff
@@ -124,6 +173,48 @@ def get_prs() -> Dict[str, Any]:
     return prs_by_branch_base
 
 
+def get_branches_with_magic_label_or_open_pr() -> Set[str]:
+    pr_infos: List[Dict[str, Any]] = []
+
+    # Grab all PRs with the magic label
+    hasNextPage = True
+    endCursor = None
+    while hasNextPage:
```
Comment (huydhn) [inline .github/scripts/delete_old_branches.py:182]: I think you could refactor the while loop here a bit into a separate function that accepts the graphql query and a lambda function to terminate the loop, i.e. `PR_WINDOW and now - convert_gh_timestamp(pr_info["updatedAt"]) > PR_WINDOW`.  The while loops to get `GRAPHQL_ALL_PRS_BY_UPDATED_AT`, `GRAPHQL_NO_DELETE_BRANCH_LABEL`, and `GRAPHQL_OPEN_PRS` look pretty similar

**Review by huydhn — APPROVED** [review]
Overall LGTM! But there are some refactoring that could be done before landing this to reduce code duplication in the script.

File: .github/scripts/delete_old_branches.py [hunk 2]
```diff
@@ -41,7 +41,52 @@
...
+        number
+        updatedAt
+        state
+      }
+    }
+  }
+}
+"""
+
+GRAPHQL_NO_DELETE_BRANCH_LABEL = """
```
[the remark this answers is not in the document]
Comment (clee2000) [reply .github/scripts/delete_old_branches.py:74]: Yep, i realized I'd have to query all the PRs to check for the string, which takes a while and uses a lot of tokens

**Pull request closed without merging**